include/boost/geometry/algorithms/detail/overlay/enrich_intersection_points.hpp

```diff
@@ -396,16 +397,36 @@ inline void enrich_intersection_points(Turns& turns,
             std::vector<indexed_turn_operation>
         >;
 
-    // From here on, turn indexes are used (in clusters, next_index, etc)
-    // and turns may not be DELETED - they may only be flagged as discarded
-    discard_duplicate_start_turns(turns, geometry1, geometry2);
+    // Turns are often used by index (in clusters, next_index, etc)
+    // and turns may therefore NOT be DELETED - they may only be flagged as discarded
 
     bool has_cc = false;
-    bool const has_colocations
-        = detail::overlay::handle_colocations
+    bool has_colocations = false;
+
+    if (BOOST_GEOMETRY_CONDITION (! is_buffer))
```

Comment:
`BOOST_GEOMETRY_CONSTEXPR` could be used instead.

Reply:
As soon as your PR is merged ;-)

Reply:
Ok, but in case you wasn't aware, this macro is already in develop:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/util/constexpr.hpp

Reply:
👍 right! Will update on Wednesday